src/main/ts/ui/Buttons.ts

```diff
@@ -0,0 +1,118 @@
+import { Editor } from "tinymce";
+
+const register = (editor: Editor): void => {
+  const onAction = () => editor.execCommand('mceEnhancedList');
+
+  editor.ui.registry.addButton('shan-enhanced-lists', {
+    type: 'button',
+    icon: 'smiley',
+    tooltip: 'shan-enhanced-list',
+    onAction: onAction
+  });
+
+  editor.ui.registry.addSplitButton('enhanced-list-unordered-list', {
+    icon: 'unordered-list',
+    tooltip: 'Enhanced unordered list',
+    presets: 'listpreview',
+    columns: 3,
+    fetch: (callback) => {
+      const items: unknown[] = [
```

Comment:
Try to avoid using `unknown` types. Make sure that you can't import the proper type from somewhere. It should be exposed through `bridge` somewhere. Don't be afraid to ask in slack for where the types might be.